primitives/session/src/lib.rs

```diff
@@ -130,3 +130,29 @@ where
 
 	Ok(())
 }
+
+/// A trait to be notified when the session changes
```

Comment:
How is this different from the other trait we discussed? should be documented to avoid further confusion.